libwallet/assets/btc/sync.go

```diff
@@ -368,19 +367,20 @@ func (asset *Asset) CancelSync() {
 	asset.syncData.wg.Add(1)
 	go asset.stopSync()
 
+	// Indicate that the sync shutdown process is fully complete.
+	asset.EndSyncShuttingDown()
```

Comment:
EndSyncShuttingDown() will call in the end of stopSync().

Reply:
Please find more information about it here: https://github.com/crypto-power/cryptopower/pull/563#issuecomment-2209125745

Reply:
I have understood exactly what you did, so why I tell you move it to stopSync() function. Because the cancel will complete only if stopSync() completes and as you can see it is in a different gouroutine, it is completely different from your implementation with dcr wallet

Reply:
@dmigwi please check comment above